app/namespaces/data_provider/data_provider_controller.py

```diff
@@ -59,13 +69,14 @@ def post(self):
         filters = data.get('filters')
 
         # Validate input payload
-        payload, status_code = validate_request_input_against_schema(data, RecordsSchema())
+        payload, status_code = validate_request_input_against_schema(data, PaginatedRecordsSchema())
         if status_code != 200:
             # If there was an error with the input payload, return the error and 422 response
             return make_response(payload, status_code)
 
         sorting_key = data.get('sorting_key')
```

Comment:
Since sorting_key, per_page, reverse and columns can all be None (based on the schema u defined), this will fail if any of these keys are not in data. We should do data.get(<keyname>, None) (e.g. data.get('sorting_key'None)) so that the variable will default to None if it doesn't exist in the payload

Reply:
Great catch, updated this for `sorting_key`, `per_page` and `reverse`. Columns isn't be allowed to take `None`, so I think we should be good for that param.